### PR TITLE
optimize: SDK bundle size by ~100kb

### PR DESCRIPTION
**Pull request by VladimirMikulic**
We use Safe Wallet SDK in our commercial Web3 application.
The main problem thus far has been huge bundle size.

After deeper investigation, I found out that inside SDK whole `ethers` package was imported.
Moreover, `ethers` itself isn't architected in the way so it can be tree shaken.

Given that SDK only uses `ethers` in a few places (message hashing and function data encoding), I've decided to migrate to alternative [`viem`](https://viem.sh/) which comes with much better TypeScript support and most importantly, the whole package is 3x smaller and optimized for tree shaking.

After testing these changes in our production, we experienced a significant drop from 504kb to 359kb in bundle size.
I expect other apps to have similar result. (100 - 150kb drop depending on bundler configuration)

### CODE REVIEW

**Comment by VladimirMikulic** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by VladimirMikulic** [comment]
recheck

**Comment by mmv08** [comment]
Hey, could you please revert all changes to the package.json and add a changeset instead? There's a comment above from the changeset bot that contains the instructions :)

**Comment by VladimirMikulic** [comment]
Appreciate the fast reply @mmv08. Changeset has been added.

**Comment by hammeiam** [comment]
Hey all, just wondering what the latest was on this PR?

**Comment by VladimirMikulic** [comment]
@hammeiam PR waiting review. Feel free to subscribe to PR to get updates as we progress.

**Review by mmv08 — APPROVED** [review]
Thanks! LGTM

**Comment by malonehedges** [comment]
when can we expect this to ship? it seems like it's been approved 2 weeks ago

**Review by malonehedges — APPROVED** [review]

**Review by tinaszheng — APPROVED** [review]

**Review by regynald — APPROVED** [review]

**Review by twofactor — APPROVED** [review]

File: packages/safe-apps-sdk/package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
   "license": "MIT",
   "dependencies": {
     "@safe-global/safe-gateway-typescript-sdk": "^3.5.3",
-    "viem": "0.3.21"
+    "viem": "1.0.0"
```
Comment (jxom) [inline packages/safe-apps-sdk/package.json:27]: /

Comment (VladimirMikulic) [reply packages/safe-apps-sdk/package.json]: Good point @jxom - fixed. Thanks!

**Comment by malonehedges** [comment]
cc @yagopv @dasanra  @DaniSomoza

**Review by yagopv — APPROVED** [review]

**Pull request merged**